src/app/pages/PerpetualPage/components/TradingChart/helpers.ts

```diff
@@ -30,6 +31,56 @@ export const config = {
   supports_time: false,
 };
 
+const addMissingBars = (bars: Bar[], candleDuration: CandleDuration): Bar[] => {
+  const candleDetails = CandleDictionary.get(candleDuration);
+  const seconds = candleDetails.candleSeconds * 1e3;
+  let newBars: Bar[] = [];
+  for (let i = 0; i < bars.length; i++) {
+    const now = new Date().getTime();
+    const latestBar = Math.floor(now / seconds) * seconds;
+    if (bars[bars.length - 1].time !== latestBar) {
+      bars.push({
+        time: latestBar,
+        open: bars[bars.length - 1].close,
+        high: bars[bars.length - 1].close,
+        low: bars[bars.length - 1].close,
+        close: bars[bars.length - 1].close,
+        volume: 0,
+      });
+    }
+    if (i === 0) {
+      newBars.push(bars[i]);
+    } else if (bars[i].time === newBars[newBars.length - 1].time + seconds) {
+      /** Is next bar in sequence */
+      console.log('IS NEXT BAR: ', bars[i]);
```

Comment:
`/** */` is a descriptive comment for classes/constants/variables/functions, otherwise please use `//` or `/* */` 

```suggestion
      /* Is next bar in sequence */
```